auth_totp/models/res_users.py

```diff
@@ -29,14 +29,20 @@ def _build_model(cls, pool, cr):
              ' right. If the button is not present, you do not have the'
              ' permissions to do this.',
     )
-    mfa_login_token = fields.Char()
-    mfa_login_token_exp = fields.Datetime()
-    trusted_device_ids = fields.One2many(
-        comodel_name='res.users.device',
-        inverse_name='user_id',
-        string='Trusted Devices',
+    trusted_device_cookie_key = fields.Char(
+        compute='_compute_trusted_device_cookie_key',
+        store=True,
     )
 
+    @api.multi
+    @api.depends('mfa_enabled')
+    def _compute_trusted_device_cookie_key(self):
+        for record in self:
+            if record.mfa_enabled:
+                record.trusted_device_cookie_key = uuid.uuid4()
```

Comment:
`record.trusted_device_cookie_key = record.mfa_enabled and uuid.uuid4() or False`

^--I'm almost sure that'll fit on one line once the nesting is reduced. Otherwise, just use an intermediary var:

```
key = ...
record.tru... = ...
```

Reply:
This doesn't really seem like a substantive improvement, especially since two lines will be needed, so I'm leaving it as is.